Lesson3a

paulcraig opened the pull request:
I believe I corrected all the formatting errors in Multiple File Parsing and Writing Files, then saved after running "Restart and Clear Output." It looks good in Jupyter Book to me.

paulcraig committed:
corrected a few typos

paulcraig committed:
first half of the file

paulcraig committed:
adding PDB files to the site

paulcraig committed:
small molecule file added

paulcraig committed:
full draft of lesson 2

paulcraig committed:
sample output file for lesson 3

paulcraig committed:
full draft of lesson 3

paulcraig committed:
sample written file from lesson 3

paulcraig committed:
sample resolutions.txt output file for lesson 3

paulcraig committed:
changed the format for the exercises

paulcraig committed:
output for the final exercise

paulcraig committed:
updated the table of contents

paulcraig committed:
changing the format for Jupyter book

paulcraig committed:
corrected typos and formatting errors

paulcraig committed:
full draft with output cleared

janash commented:
This branch should not have notebooks or commits for previous lessons. It should have the notebook for lesson 3 only. It is likely that the notebook for lesson 2 (file parsing) will change before being merged and having it on this branch as well will introduce conflicts.

You may have data files which will be used in both lessons. However, since these files are static, having them present on each pull request will not cause as much of an issue. I do recommend that you only add files as they are needed for lessons, however.

janash commented on biochemist-python/chapters/Processing_Multiple_Files_and_Writing_Files.ipynb:
```diff
@@ -0,0 +1,427 @@
+    ":class: dropdown\n",
+    "\n",
+    "You can use the str.split function introduced in the last lesson, and split at the '.' character.\n",
+    "    \n",
```
Need to show code block here.

paulcraig replied:
I did not know how to remove files, so I looked for instructions on [StackOverflow ](https://stackoverflow.com/questions/2047465/how-can-i-delete-a-file-from-a-git-repository/17477961#17477961)and did this.

1. On Github, lesson3a branch, I deleted the unnecessary files.
2. I went to my lesson3a branch and did a pull request (git pull main lesson3a). 

I won't do this to any other branches until I have confirmed with you that I did no damage.

Also, as I looked at the file, I thought that it was redundant have two folders of protein structures: PDB_files and enzymes. So I eliminated the enzymes folders and am using the PDB_files folder for this notebook.

paulcraig replied:
I moved the code block below this cell into the solution.

janash commented on biochemist-python/chapters/Processing_Multiple_Files_and_Writing_Files.ipynb:
```diff
@@ -0,0 +1,427 @@
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "split_filename = file_name.split('.')\n",
```
This is the exercise solution. Should be repeated in expandable section.

paulcraig replied:
Done.

janash commented on biochemist-python/chapters/Processing_Multiple_Files_and_Writing_Files.ipynb:
```diff
@@ -0,0 +1,427 @@
+    "    molecule_name = split_filename[0]\n",
+    "\n",
+    "    # Read the data\n",
+    "    outfile = open(f,'r')\n",
```
Will want to switch this to `with` if `with` is used in previous lesson.

paulcraig replied:
Done. Please review the text in Printing to a File. I rewrote much of it and made it all markdown, rather than having executable code. Is this too long a markdown section?

janash commented on biochemist-python/chapters/Processing_Multiple_Files_and_Writing_Files.ipynb:
```diff
@@ -0,0 +1,427 @@
+   "source": [
+    "After you run this command, look in the directory where you ran your code and find the \"resolutions.txt\" file.  Open it in a text editor and look at the file.\n",
+    "\n",
+    "In the file writing line, notice the `\\n` at the end of the line.  This is the newline character.  Without it, the text in our file would just be all smushed together on one line.  Also, the `filehandle.close()` command is very important.  Think about a computer as someone who has a very good memory, but is very slow at writing.  Therefore, when you tell the computer to write a line, it rem ...
```
Can remove this if we use `with`

paulcraig replied:
I just moved the text about the filehandle.close() to the Printing to a File section (making it even longer, unfortunately).

janash commented on biochemist-python/chapters/Processing_Multiple_Files_and_Writing_Files.ipynb:
```diff
@@ -0,0 +1,427 @@
+   "metadata": {},
+   "source": [
+    "<div class=\"exercise-this-is-a-title exercise\">\n",
+    "<p class=\"exercise-title\">Exercise</p>\n",
```
Consider calling this something other than "exercise" if you think it will take a while.

paulcraig replied:
I changed _Exercise_ to _Project_.

janash commented on biochemist-python/chapters/Processing_Multiple_Files_and_Writing_Files.ipynb:
```diff
@@ -0,0 +1,427 @@
+    "filenames = glob.glob(file_location)\n",
+    "# print(filenames)  #Once you're sure it's working, comment this line out.\n",
+    "\n",
+    "# *Loop through the lines in the files*\n",
```
Nice exercise/project. I like that it uses multiple files (the subject of the lesson). I also didn't know about enzyme commission classes.

Note that if we switch to opening files with `with`, this solution should be adjusted.

paulcraig replied:
Just switched it over to using _with open_. Is there a way for me to keep the actual solution cell with the notebook, but hide it from the user? I'm just thinking it would be nice to have in place in case python changes and we need to rewrite the solution.

janash reviewed (changes requested):
Please only have files related to work in a pull request in the pull request. This should have the notebook and relevant files for lesson 3 only.

paulcraig committed:
Delete File_Parsing.ipynb

paulcraig committed:
Delete introduction.ipynb

paulcraig committed:
Delete remdesivir.pdb

paulcraig committed:
Delete remdesivir.sdf

paulcraig committed:
Delete remdesivir.xyz

paulcraig committed:
Delete 1vxh.pdb

paulcraig committed:
Delete 4eyr.pdb

paulcraig committed:
Delete 4ezi.pdb

paulcraig committed:
Delete 6l35.pdb

paulcraig committed:
Delete 6lsv.pdb

paulcraig committed:
Delete 6vqs.pdb

paulcraig committed:
Delete 7bfv.pdb

paulcraig committed:
Delete 7bgj.pdb

paulcraig committed:
Delete 7dno.pdb

paulcraig committed:
Delete 7l1f.pdb

paulcraig committed:
Delete 7tim.pdb

paulcraig committed:
Delete remdesivir.xyz

paulcraig committed:
made all changes; ran and cleared output

paulcraig committed:
adding all enzymes to the PDB_files folder

paulcraig committed:
Delete 2pkr.pdb

paulcraig committed:
Delete 3iva.pdb

paulcraig committed:
Delete 3vnd.pdb

paulcraig committed:
Delete 5eu9.pdb

paulcraig committed:
Delete 5veu.pdb

paulcraig committed:
Delete 6zt7.pdb

paulcraig committed:
Delete 7tim.pdb

paulcraig committed:
updated TOC

paulcraig committed:
changes made, restarted and cleared output

paulcraig committed:
Delete Processing_Multiple_Files_and_Writing_Files.ipynb

Something wrong with my git operations. I am deleting this file on Github and will try to send it from my local repository.

paulcraig committed:
written file from the main notebook

paulcraig committed:
current state of lesson 3

paulcraig committed:
copy of lesson 3

janash committed:
Merge branch 'lesson3a' of https://github.com/paulcraig/python-scripting-biochemistry into paulcraig-lesson3a

janash committed:
slight clean up of notebook, delete unneccessary file

janash commented:
Hi @paulcraig - I did some slight clean up on this pull request and pushed (I can actually edit your pull requests 😄 ). A few things:

- I did say that only files related to the current lesson should be on the branch. This is only true if the other files are not already on the main branch. The `introduction` notebook is already on the main branch, so should be left on this one. Otherwise, having to delete it will result in the file being deleted when this pull request is merged.
- I changed one case of using `open` and `close` for opening a file with the `with` syntax. I just think it's better to be consistent and to always use the same thing.
- I deleted the files that are produced from this notebook. We don't really need them in the repo unless we will be reading them again.
